src/main/java/io/vertx/howtos/metrics/MainVerticle.java

```diff
@@ -0,0 +1,100 @@
+package io.vertx.howtos.metrics;
+
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.config.MeterFilter;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.eventbus.Message;
+import io.vertx.ext.web.Router;
+import io.vertx.micrometer.MicrometerMetricsOptions;
+import io.vertx.micrometer.PrometheusScrapingHandler;
+import io.vertx.micrometer.VertxPrometheusOptions;
+import io.vertx.micrometer.backends.BackendRegistries;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+public class MainVerticle extends AbstractVerticle {
+
+  // tag::greetings[]
+  private static final String[] GREETINGS = {
+    "Hello world!",
+    "Bonjour monde!",
+    "Hallo Welt!",
+    "Hola Mundo!"
+  };
+  // end::greetings[]
+
+  @Override
+  public void start() {
+    // tag::consumer[]
+    vertx.eventBus().consumer("greetings", msg -> {
+      // Simulate processing time between 20ms and 100ms
+      long delay = ThreadLocalRandom.current().nextLong(80) + 20L;
+      vertx.setTimer(delay, l -> {
+        // Choose greeting
+        String greeting = GREETINGS[ThreadLocalRandom.current().nextInt(GREETINGS.length)];
+        msg.reply(greeting);
+      });
+    });
+    // end::consumer[]
+
+    // tag::router[]
+    Router router = Router.router(vertx);
+    // end::router[]
+    // tag::handler[]
+    router.get("/greeting").handler(rc -> {
+      vertx.eventBus().<String>request("greetings", null, ar -> {
```

Comment:
How about using `Future` here? (nothing wrong at all with the callback approach)